docker-compose.yml

```diff
@@ -351,7 +353,65 @@ services:
       - .:/arrow:delegated
       - ${DOCKER_VOLUME_PREFIX}ubuntu-ccache:/ccache:delegated
     command: *cpp-command
+  
+  ubuntu-cpp-static:
+    # Usage:
+    #   docker-compose build ubuntu-cpp-static
+    #   docker-compose run --rm ubuntu-cpp-static
+    # Parameters:
+    #   ARCH: amd64, arm64v8, s390x, ...
+    #   UBUNTU: 18.04, 20.04
+    image: ${REPO}:${ARCH}-ubuntu-${UBUNTU}-cpp-static
+    build:
+      context: .
+      dockerfile: ci/docker/ubuntu-${UBUNTU}-cpp.dockerfile
+      cache_from:
+        - ${REPO}:${ARCH}-ubuntu-${UBUNTU}-cpp-static
+      args:
+        arch: ${ARCH}
+        base: "${ARCH}/ubuntu:${UBUNTU}"
+        clang_tools: ${CLANG_TOOLS}
+        llvm: ${LLVM}
+        gcc_version: ${GCC_VERSION}
+    shm_size: *shm-size
+    cap_add:
+      - SYS_ADMIN
+    devices:
+      - "/dev/fuse:/dev/fuse"
+    security_opt:
+      - apparmor:unconfined
+    ulimits: *ulimits
+    environment:
+      <<: *ccache
+      ARROW_HOME: /arrow
+      ARROW_DEPENDENCY_SOURCE: BUNDLED
+      LIBARROW_MINIMAL: "false"
+      ARROW_MIMALLOC: "ON"
+    volumes: *ubuntu-volumes
+    command: /bin/bash -c "
+        cd /arrow && r/inst/build_arrow_static.sh"
```

Comment:
This change seems to introduce new `R` builds, not `C++` builds.

Reply:
It could be defined in tasks.yml like:

```yml
 test-ubuntu-18.04-r-static:  # or something related to R
    ci: github
    template: docker-tests/github.linux.yml
    params:
      env:
        UBUNTU: 18.04
      flags: "-e ARROW_DEPENDENCY_SOURCE=BUNDLED -e ARROW_HOME=/arrow -e LIBARROW_MINIMAL=false"
      command: "/arrow/r/inst/build_arrow_static.sh"
      image: ubuntu-cpp
```

Reply:
> This change seems to introduce new `R` builds, not `C++` builds.

Well it is a libarrow build for use in the R package, so both in a way?
The task itself needs a bit more involved [template](https://github.com/assignUser/arrow/blob/ebf0e00db702bef2f00621318df92f50cf3711e9/dev/tasks/r/github.nightly.yml#L1) than most jobs (at least currently moving it over 1:1 as mentioned above, we can probably find a way to be more efficent later on)